azure-ts-dynamicresource/index.ts

```diff
@@ -0,0 +1,96 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as azure from "@pulumi/azure";
+
+import { CDNCustomDomainResource } from "./cdnCustomDomain";
+
+/**
+ * The location where our resource group and the resources under it will be created.
+ *
+ * To externalize this value, and make this configurable across environments/stacks,
+ * learn more at https://pulumi.io/reference/config/.
+ */
+const location = "West US";
+
+/** The custom domain host of the CDN endpoint. */
+export let cdnCustomDomainResource: CDNCustomDomainResource | undefined;
+
+// Create an Azure Resource Group
+const resourceGroup = new azure.core.ResourceGroup("resourceGroup", {
+    location: location,
+});
+
+// Create an Azure resource (Storage Account) so we can point the CDN endpoint to it.
+const storageAccount = new azure.storage.Account("storageAccount", {
+    resourceGroupName: resourceGroup.name,
+    accountTier: "Standard",
+    accountReplicationType: "LRS",
+});
+
+/**
+ * Create a Blob container in the storage account,
+ * to store any static files. The CDN endpoint will be pointed at the
+ * endpoint for this blob container.
+ */
+const blobContainer = new azure.storage.Container("blob-container", {
+    resourceGroupName: resourceGroup.name,
+    storageAccountName: storageAccount.name,
+    // Make each "blob" in the container publicly accessible.
+    // DO NOT set this property if you are going to store sensitive files!
+    containerAccessType: "blob",
+});
+
+const cdnProfile = new azure.cdn.Profile("cdn-profile", {
+    resourceGroupName: resourceGroup.name,
+    // Choose an appropriate SKU to use.
+    // https://docs.microsoft.com/en-us/azure/cdn/cdn-features
+    sku: "Standard_Akamai",
+});
+
+const cdnEndpoint = new azure.cdn.Endpoint("my-cdn-endpoint", {
+    name: "my-cdn-endpoint",
+    resourceGroupName: resourceGroup.name,
+    profileName: cdnProfile.name,
+    isHttpsAllowed: true,
+    isHttpAllowed: false,
+    isCompressionEnabled: true,
+    originHostHeader: storageAccount.primaryBlobHost,
+    contentTypesToCompresses: [
```

Comment:
aside: contentTypesToCompresses is a bizarro name.  "Compresses"??

Reply:
Yes. Unfortunately, that's how the terraform property is named. It is a typo in the provider itself.